### DISCUSSION TITLE
Videos via unparenting.app

### DISCUSSION THREAD

**Discussion by Huk-Fuk** [aclap-dev/video-downloadhelper#223] [Help and Q&A]
### Your question

Hi,
I searched unsucesfull for keywords unparenting and below nmentioned websites.

since couple of months I am unable to download videos coming via unparentig.app (https://video.unparenting.app/cz-we-zn6/cz-we-zn6_1080p/00000/cz-we-zn6_1080p_00009.ts). There is a JS on https://cdn.nevychova.cz/video-player/js/video.min.js
VDH doesn't recognize the site contains video.

The exact website with the webinar is available only today until midnight (27.Feb CET): https://www.nevychova.cz/zdarma/cz-we-zn7/vysilani-v-283/?_nvu=e6e8c3dd-530a-4e65-be5c-b4d2e3f19c9a&utm_source=webinar&utm_medium=email&utm_campaign=cz-we-zn7_vysilani1_vysilani

Thank you

### Addon & CoApp version and details

Win10 x64
FF 115.8.0esr (64-Bit)
Video DownloadHelper v7.6.6
VdhCoApp 2.0.19

**Comment by jcfrog** [comment]
Hi,
Is there any reason why you're sticking with such an old version of VDH?

**Reply by GrampaWildWilly** [reply]
I was going to ask the same thing.

**Reply by Huk-Fuk** [reply]
Ah, it did not update automatically... my fault. Now it is v8.2.1.1 (stable), but:

moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1
receive@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:13157
Zp</callCatchAppNotFound/</<@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:42626

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 123.0 64-bit, licensed VDH 8.2.1.3a1 beta, CoApp 2.0.19.  I should also point out that I do not work for VDH.  I'm just another user, like you.

The first 2 links you provided are pretty much no help at all.  The first one is a ts.  It looks like the ninth one, maybe the tenth if they start numbering them from 0.  It is just one chunk of the stream that makes up your video.  The second link is a js.  If I'm not mistaken, that is a javascript.  I can spell javascript, as you can see.  But don't ask me to explain it.  I have no idea what it is.  Most web sites have lots of them.  They are irrelevant to VDH.

I visited your web site using the third link you provided.  That is useful.  This is what I got:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/5cab96e6-a44b-4fe7-af9e-30c9492bcbd4)

I know these images are hard to read.  Just click on an image.  GitHub will open the image in a separate browser window.  That might still be a bit small but you can use the browser's magnifier to make things legible.

I launched the download, as shown there.  While it was in flight, VDH told me this on its status menu:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f43e2ae0-4dcc-48bd-8490-838338f67e96)

At the same time, Resource Monitor told me this:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/53eae4b5-ac67-44f7-b3d9-f0563bbb1d55)

When the download completed, I had these results:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/60777dda-6bab-486c-a610-36ba9660ba85)

Here it is playing in VLC:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/5ce44000-6dcb-427a-aaa1-7071dc44114b)

I didn't sit & watch this.  Not only is it over an hour & a half long, but she's speaking Czech, a language I don't speak.  I sampled it at intervals.  You can see it's playing just fine.  You'll have to take my word for it that the audio was just as fine, all the way to the end.

It occurred to me to check whether there are captions with this.  So I scrounged around in the Network Monitor.  Normally for an HLS stream, which is how VDH recognized this as you can see in my first image above, there will be an HLS master manifest, a file with a file extension of m3u8, sent from the web site to your browser.  It will be of type x-mpegurl.  I didn't find one of those.  But I did find this:

![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7aabb3ed-d3f3-4455-b0dd-5c0a279dab06)

Its type is plain, not x-mpegurl.  That is an extremely non-standard way of supplying an HLS master manifest.  In any case, I used the popup context menu on that manifest to execute the Copy URL function.  This put the URL of that manifest into my system clipboard.  From there, I pasted that into an invocation of ffprobe.  This is what I got.

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14417617/ffprobe.txt)

There are no subtitles.  I do note, however, the errors on streams 3 & 5.  Those are the audio tracks for the 960x540 & 1280x720 variants you can see in my first image above.  The report shows there are 0 channels for those tracks.  I suspect that if you downloaded either of those 2 variants, you would get either a video without audio or simply a failure to download.  Whatever the case, ffprobe appears to be telling us that those audio tracks are damaged in some way.  Fortunately, the 1920x1080 variant is fine.  That's the one I downloaded & played.

I find it curious that you are using the very latest CoApp but you are using a woefully out of date VDH.  That one is nearly a year old.  Also, you are using another piece of woefully out of date software: Firefox.  You're on Windows 10, not Windows 7.  That version of Firefox is the last one Mozilla is permitting on Windows 7, & they're going to stop maintaining it in any way at all later this year.  There should be nothing stopping you from getting the latest Firefox, which at the moment is the one I'm using, as you can read at the beginning of this post.

I would put forward the hypothesis that your terribly out of date software is what is preventing you from downloading this video.  I had no problem with it.  You should try what I'm using.  Click the link labelled Wiki at the top of this page.  Then scan down the Table of Contents on the right side of the Wiki page.  Look for the chapter on the latest beta version of VDH.  That's what you should be running.  And you ought to get on the latest Firefox, as well.  I have every confidence that will cure your problem here as well as a number of other problems you may be having with other content.  It will probably also keep you from falling into problems with things you haven't even downloaded yet.

**Reply by Huk-Fuk** [reply]
Nice, I just updated the VDH (then I see the downloadable resolution versions), but got this error:

oz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1
receive@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:13157
Zp</callCatchAppNotFound/</<@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:42626

Any idea, what it should be?

**Reply by paulrouget** [reply]
Do you have a link to a specific public page where you can reproduce this at the moment?

**Reply by GrampaWildWilly** [reply]
Have you tried the 8.2.1.3a1 beta that I used in my test above?  It worked for me.  How about for you?

**Reply by Huk-Fuk** [reply]
So, I updated the DWH (my failure, automat.updats were switched off) v8.2.1.1 as well as with v8.2.2.6a1 (beta) ends with the same failure:
_"moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1
receive@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://4059f5be-5c43-4bd0-8b96-88726002f47e/background/main.js:1:42624"_

here is a today functional link: https://www.nevychova.cz/zdarma/cz-we-dp5/vysilani-v-201/?_nvu=e6e8c3dd-530a-4e65-be5c-b4d2e3f19c9a&utm_source=webinar&utm_medium=email&utm_campaign=cz-we-dp5_vysilani1_vysilani

**Reply by GrampaWildWilly** [reply]
It is not your failure that automatic updates were switched off.  There is a problem whereby Firefox can spontaneously perform an automatic update of VDH while you happen to be in the process of downloading something with VDH.  This will interrupt the download & leave you with an unplayable file.  You'll have to start the download over from the beginning.  So until this issue is addressed, the proper way to do things it to turn off automatic updates & manually check for new versions of VDH.

In addition, the link to the beta for Firefox in the Wiki is not the latest.  There is a problem that Paul is working on.  In order to update VDH to the latest beta, you need to go into the Firefox settings & click the Check for Update button yourself.  And you need to remember to do that every day or 2.  Yes, it's less convenient, but it's necessary because of the issue of Firefox updating VDH even if it's busy downloading something.

And please call it VDH.  People would have no idea what DWH is.  It's VDH.  Please.  VDH.

**Reply by Huk-Fuk** [reply]
OK for sure VDH, to fast fingers, sorry. But what about the error message?

**Reply by GrampaWildWilly** [reply]
I don't know.  The latest link you posted is behind a user ID & password so I can't access it.  But that was a week ago so maybe it's expired anyway.  You might post the Hit Details of something behind a user ID & password.  That would be to cascade out the secondary menu for a variant on the main VDH menu, then click Details.  That gives you the Hit Details for that variant.  Copy/paste the entire page here.  Don't post an image.  Copy/paste the text of the Hit Details into a post here.  Sometimes that can let somebody who doesn't have a login get behind the login.  Sometimes.  I'm puzzled that you are seeing this error.  I was running the latest beta & it worked for me 3 weeks ago.  There's been a new Firefox & a few new VDH betas since I ran my successful test above.  If you're on all the latest bits of software, I don't have any ideas.  Remember.  I don't work for VDH.  I don't have access to the code.  Even though the CoApp is open source, I'm not qualified to make changes in that.  So I don't know what the issue is here.

**Reply by Huk-Fuk** [reply]
All fine it was either proxy or VPN - without both works fine,
Thank you all
